DemoApp/app/screens/DataScreen.js

```diff
@@ -54,10 +54,26 @@ export default class DataScreen extends Component {
   async refreshToggle() {
     const dataEnabled = await Data.isEnabled();
     this.setState({ dataEnabled });
+    if (!this.state.loadingData) {
+      const documents = await this.listDocuments(this.state.partition);
+      this.setState({ documents, loadingData: false });
+    }
   }
 
-  setCreateDocModalVisible(visible) {
+  async setCreateDocModalVisible(visible) {
```

Comment:
The `setCreateDocModalVisible` method and how the method is called is very confusing. I suggest we refactor the code to make it easier to follow.
 
In the current PR, when we successfully created the new document, `setCreateDocModalVisible` is called with  `!createDocModalVisible` parameter (https://github.com/microsoft/appcenter-sdk-react-native/pull/762/files#diff-5c743115ed0a0fcdd754f9eac490cb37R290). 

The confusing parts are:
-  the method name `setCreateDocModalVisible` doesn't make sense if the implementation switch whatever the visibility state between true and false.
- this method only calls `setState` when `visible` parameter is false, which is the reverse value of `createDocModalVisible`.

Reply:
I like the refactored code 👍